src/core.js

```diff
@@ -272,6 +272,16 @@ $.extend( $.validator, {
 				this.element( element );
 			}
 		},
+		oninput: function( element ) {
+			if ( element.name in this.submitted || element.name in this.invalid ) {
+				this.element( element );
+			} else {
+				return;
+			}
+		},
+
+		// DEPRECATED
+		// Please use `oninput` instead.
 		onkeyup: function( element, event ) {
```

Comment:
Sorry if I misunderstood your question.

> can we get rid of keyup handler in browsers which support the input event?

Yes we can.

> When exactly will the input event be triggered? On each keystroke?

Allow me to point to specs, see:
* https://developer.mozilla.org/en-US/docs/Web/Events/input
* https://w3c.github.io/editing/input-events.html